template/{{.input_root_dir}}/.azure/devops-pipelines/deploy-cicd.yml.tmpl

```diff
@@ -0,0 +1,93 @@
+      persistCredentials: true
+      clean: true
+      displayName: 'Checkout & Build.Reason: $(Build.Reason) & Build.SourceBranchName: $(Build.SourceBranchName)'
+
+    # Install Databricks CLI
+    - script: |
+        curl -fsSL https://raw.githubusercontent.com/databricks/setup-cli/{{template `cli_version` .}}/install.sh | sh   
+      displayName: 'Install Databricks CLI'          
+
+    # Extract Databricks version to test installation
+    - script: |
+        databricks version
+      displayName: 'Get Databricks CLI version'
+
+    # Install jq
+    - script: |
+        sudo apt-get install jq
+      displayName: 'Install jq'
+
+    # Unzip Bundle and Append Parameters to Input JSON
+    - script: |
+        tar -xzvf cicd.tar.gz
+        INPUT_CLOUD_1=$(jq -r '.input_cloud' cicd_params.json)
+        INPUT_CLOUD_2=$(jq -r '.input_cloud' {{`${{ parameters.ProjectName }}`}}/project_params.json)
+        if [ "$INPUT_CLOUD_1" != "$INPUT_CLOUD_2" ]; then
+          printf "Error: CICD cloud '%s' does not match project cloud '%s'\n" "$INPUT_CLOUD_1" "$INPUT_CLOUD_2"
+          exit 1
+        fi
+        printf '%s' "$(jq '. += {"input_project_name":"{{`${{ parameters.ProjectName }}`}}"}' cicd_params.json)" > cicd_params.json
+        printf '%s' "$(jq -s '.[0] + .[1]' cicd_params.json {{`${{ parameters.ProjectName }}`}}/project_params.json)" > cicd_params.json
+      displayName: 'Extract CICD Bundle and Update Parameters'        
+
+    # Update databricks.yml
+    - script: |
+        echo -e "  staging:\n    workspace:\n      host: {{template `databricks_staging_workspace_host` .}}\n\n  prod:\n    workspace:\n      host: {{template `databricks_prod_workspace_host` .}}\n\n  test:\n    workspace:\n      host: {{template `databricks_staging_workspace_host` .}}" >> {{`${{ github.event.inputs.project_name }}`}}/databricks.yml
+      displayName: 'Update databricks.yml'
+
+    # Initialize CICD Bundle
+    - script: |
+        databricks bundle init ./cicd --config-file "cicd_params.json"
+      displayName: Initialize CI/CD Bundle
+      env:
+        DATABRICKS_HOST: {{template `databricks_staging_workspace_host` .}}
+        {{ if (eq .input_cloud `aws`) -}}
+        DATABRICKS_TOKEN: $(STAGING_WORKSPACE_TOKEN)
+        {{- else if (eq .input_cloud `azure`) -}}
+        ARM_TENANT_ID: $(STAGING_AZURE_SP_TENANT_ID)
+        ARM_CLIENT_ID: $(STAGING_AZURE_SP_APPLICATION_ID)
+        ARM_CLIENT_SECRET: $(STAGING_AZURE_SP_CLIENT_SECRET)
+        {{- end }}
+
+    # Create Branch and Commit CICD Bundle
+    - script: |
+        git config --global user.name "Deploy CICD Bot"
+        git config --global user.email "noreply-cicd-bot@databricks.com"
+        git checkout -b add-cicd-for-{{`${{ parameters.ProjectName }}`}}
+        git add .azure
+        git commit -m "Add CICD for {{`${{ parameters.ProjectName }}`}}"
+        git push origin add-cicd-for-{{`${{ parameters.ProjectName }}`}}
+      displayName: Push CICD Bundle to a New Branch
```

Comment:
Bit concerned about adding `--force` on the off-chance someone has the same branch or has force pushes disabled on their repository. Seems a bit weird to need to do add this flag

Reply:
Added `--force-with-lease` to prevent the overwriting possibility, and someone can just update the pipeline to remove this if they have force disabled on the whole repo/org